docs/docs/adrs/adr-011-improving-test-confidence.md

```diff
@@ -0,0 +1,208 @@
+---
+sidebar_position: 12
+title: ADR Template
+---
+# ADR 11: Improving testing and increasing confidence
+
+## Changelog
+* 2023-08-11: Proposed, first draft of ADR.
+
+## Status
+
+Proposed
+
+## Context
+
+Testing, QA, and maintenance of interchain-security libraries is an ever-evolving area of software engineering we have to keep incrementally improving.
+
+The purpose of the QA process is to catch bugs as early as possible. In an ideal development workflow a bug should never reach production.
+
+A bug found in the specification stage is a lot cheaper to resolve than a bug discovered in production (or even in testnet).
+
+Ideally, all bugs should be found during the CI execution, and we hope that no bugs will ever even reach the testnet (although nothing can replace actual system stress test under load interacting with users).
+
+During development and testnet operation the most commonly found types of bugs are listed below.
+- improper iterator usage
+- unbounded array access/iteration
+- improper input handling and validation
+- improper cached context usage
+- non-determinism check (improper use of maps in go, relying on random values)
```

Comment:
hi, for these common bugs, if there is a linked issue or some docs can be linked here? or add a short description?